### DISCUSSION TITLE
I am not able to make custom authorizer work

### DISCUSSION THREAD

**Discussion by piotrwalczak1** [serverless/serverless#10515] [General]
I have the following serverless.yaml:

    getSth: 
        handler: src/handlers/getSth.getSth
        events:
            - http:
                path: getSth
                method: get
                cors: true
                private: true
                authorizer: authorizerFunc
    authorizerFunc:
        handler: src/handlers/authorizer.authorizer
getSth handler:

    module.exports.getSth = async (event, context) => {
    const response = {
        statusCode: 200,
        body: JSON.stringify({message: "nice you can call this});
    }
    return response;
    }
authorizerFunc:

    module.exports.authorizer = async (event, context) => {
         console.log('i will fail your authorization');
         let response = {
         isAuthorized: false,
         context: {
           stringKey: "value",
           numberKey: 1,
           booleanKey: true,
           arrayKey: ["value1", "value2"],
           mapKey: { value1: "value2" },
        },
      };
    return response;
     }
That results in getting respons 200 in spite of the fact authorizer should not allow to execute that getSth function. Also console log 'I will fail your authorization' is not logged.

What am I doing wrong ?

**Comment by pgrzesik** [comment]
Hello @piotrwalczak1 - how are you calling your function? I've tested similar setup and it works correctly. Please note that if you're invoking your Lambda directly then authorizer won't be called as it's configured on API GW level.

**Reply by piotrwalczak1** [reply]
Hi,
Ok, it work. I have been testing it with sls offline --noAuth. And it was not calling lambda authorizer then.
@pgrzesik , I have one more question? Which is (according to you) better ? Use lambda authorizer or token middleware to validate the token ? What do you think, because this is the decision design I have to make?

**Reply by pgrzesik** [reply]
Hey @piotrwalczak1 - it's hard to answer such question in vacuum and without additional context, I'm also not 100% sure what exactly do you mean by `token middleware`.

**Reply by piotrwalczak1** [reply]
@pgrzesik by token middleware I mean to validate if someone who is calling api has the valid jwt token. I can do this by implementing lambda function to validate a jwt token and point to that function in serverless yaml, or I can to that using some middleware by validating the token using some middleware like middy from npm. (that would decorate the original function and check jwt token before executing that function).

**Reply by pgrzesik** [reply]
If you're using JWT you don't have to use custom Lambda authorizers but instead can use native JWT authorizers on Api Gateway level: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-jwt-authorizer.html
